index.d.ts

```diff
@@ -1,25 +1,12 @@
-export class InvalidTokenError extends Error {}
+import { InvalidTokenError, JwtDecodeOptions, JwtHeader, JwtPayload } from "./common";
 
-export interface JwtDecodeOptions {
-  header?: boolean;
+declare namespace jwtDecode {
+  export { InvalidTokenError, JwtDecodeOptions, JwtHeader, JwtPayload }
 }
 
-export interface JwtHeader {
-  typ?: string;
-  alg?: string;
-}
-
-export interface JwtPayload {
-  iss?: string;
-  sub?: string;
-  aud?: string[] | string;
-  exp?: number;
-  nbf?: number;
-  iat?: number;
-  jti?: string;
-}
-
-export default function jwtDecode<T = unknown>(
+declare function jwtDecode<T = unknown>(
   token: string,
   options?: JwtDecodeOptions
-): T;
+): T
+
+export = jwtDecode;
```

Comment:
It looks like this change is breaking existing functionality.

Take the following example:

```ts
import decode, { JwtHeader, JwtPayload } from 'jwt-decode';

const token = '...';
const header = decode<JwtHeader>(token, {header: true});
const body = decode<JwtPayload>(token, {header: false});

console.log(header.typ);
console.log(header.alg);
console.log(body);
```
The above will work fine with the current version in master, but with the changes in this PR, it fails:

```
TS1259: Module '"C:/Development/Playground/jwt-decode-test/node_modules/jwt-decode/index"' can only be default-imported using the 'esModuleInterop' flag
```
Once resolved the above, the same snippet still fails:

```
TS2614: Module '"jwt-decode"' has no exported member 'JwtPayload'. Did you mean to use 'import JwtPayload from "jwt-decode"' instead?
```

Reply:
The error @frederikprijck mentioned here is expected behaviour.

If a TypeScript project is compiled to CommonJS, then this is the correct way to import a CJS dependency:

```ts
import jwtDecode = require('jwt-decode');
// compiles to
// const jwtDecode = require('jwt-decode');
```

If the user wishes to use a faux ESM default import to import a library in a code base that gets converted to CJS, then the user needs to enable `esModuleInterop` in their compiler options. The user can then use a default import.

```ts
import jwtDecode from 'jwt-decode'
// compiles to
// "use strict";
// var __importDefault = (this && this.__importDefault) || function (mod) {
//     return (mod && mod.__esModule) ? mod : { "default": mod };
// };
// exports.__esModule = true;
// var index_js_1 = __importDefault(require("./index.js"));
// (0, index_js_1["default"])('');
```

If the users wishes to use a CJS project with a module bundler, they need to enable `allowSyntheticDefaultImports`.